### DISCUSSION TITLE
non-existent service after upgrade from 6.9 to X

### DISCUSSION THREAD

**Discussion by Bradeskojest** [pimcore/pimcore#11295] [Q&A / Support]
By following the guide for updating the Pimcore, I am now stuck in this error, which happens after a command:
`COMPOSER_MEMORY_LIMIT=-1 composer update`

```
Script Pimcore\Composer::clearCache handling the pimcore-scripts event terminated with an exception


  [RuntimeException]
  An error occurred when executing the "/usr/local/bin/php --php-ini=/usr/local/etc/php/php.ini bin/console --ansi cache:clear --no-warmup" command:

  Exit code: 255

  Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException {#7448
    -id: "security.authentication.provider.dao"
    -sourceId: null
    -alternatives: []
    #message: "You have requested a non-existent service "security.authentication.provider.dao"."
    #code: 0
    #file: "./vendor/symfony/dependency-injection/ContainerBuilder.php"
    #line: 996
    trace: {
      ./vendor/symfony/dependency-injection/ContainerBuilder.php:996 { …}
      ./vendor/symfony/dependency-injection/ContainerBuilder.php:1028 { …}
      ./vendor/pimcore/pimcore/bundles/CoreBundle/DependencyInjection/Compiler/PasswordFactoryDecoratorPass.php:35 { …}
      ./vendor/symfony/dependency-injection/Compiler/Compiler.php:82 { …}
      ./vendor/symfony/dependency-injection/ContainerBuilder.php:757 { …}
      ./vendor/symfony/http-kernel/Kernel.php:548 { …}
      ./vendor/pimcore/pimcore/lib/Kernel.php:237 { …}
      ./vendor/symfony/http-kernel/Kernel.php:789 { …}
      ./vendor/symfony/http-kernel/Kernel.php:128 { …}
      ./vendor/pimcore/pimcore/lib/Kernel.php:216 { …}
      ./vendor/pimcore/pimcore/lib/Bootstrap.php:255 { …}
      ./vendor/pimcore/pimcore/lib/Bootstrap.php:60 { …}
      ./bin/console:34 {
        › /** @var \Pimcore\Kernel $kernel */
        › $kernel = \Pimcore\Bootstrap::startupCli();
        › $application = new \Pimcore\Console\Application($kernel);
      }
    }
  }

```

**Comment by jdreesen** [comment]
Did you merge/copy Pimcore's firewall config into your own? 
See: https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/Updating_Pimcore/V6_to_V10.html#page_Adapt-Security-config

**Reply by Bradeskojest** [reply]
I was looking into this, but it was not clear what I need to do. My firewall rules were empty. I am re-doing now all the steps again, from the backup, and will look into this again.

**Reply by Bradeskojest** [reply]
This returns nothing, before update: `bin/console debug:config security firewalls`

**Reply by Bradeskojest** [reply]
I ended up in the same error loop:/

**Comment by John5** [comment]
I spent a lot of time with these errors too and eventually gave up. The official migration guide refers to all kinds of lists of files that need to be moved, renamed and hundreds of lines of arbitrary configuration lines that need to be pasted from some code repository and then you get endless untracable error messages. A huge waste of time.

It was easier to start with a fresh Pimcore 10 installation and copy my code into the new structure. Then I pointed it to the old database and admin interface then complains about some missing columns but at least the SQL errors are clear about what the problem is and easy enough to fix.

**Comment by TsumeMS** [comment]
Hello,
First I upgraded from 6.3 to 6.9 without problems and then to 10 version and I have the same error. Anyone have a solution to this problem?
